tests/test_events/handler/addTab.test.js

```diff
@@ -0,0 +1,119 @@
+        fs.mkdirSync(tabsRepo, { recursive: true });
+        createSystemDirectories();   
+        console.log(`Creating directories: repoPath=${repoPath}, tabsRepo=${tabsRepo}`);     
+        jest.clearAllMocks();
+        untildify.mockReturnValue(baseRepoPath);
+        const returnedPath = untildify();
+        console.log(`untildify returned: ${returnedPath}`);
+    });
+
+    afterEach(() => {
+        fs.rmSync(repoPath, { recursive: true, force: true });
+        fs.rmSync(baseRepoPath, { recursive: true, force: true });
+    });
+
+    test('untildify should return the correct mocked path', () => {
+        untildify.mockReturnValue(baseRepoPath);
+        expect(untildify()).toBe(baseRepoPath);
+    });
+    
+    // Skips case when repo exists
+    test("Should be skipped if repo does not exist",() => {
+        const handler = new tabEventHandler();
+        handler.handleTabChangeEvent();
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)   
+        });
+
+    //  Skips cases where repo exists, but is not connected
+    test("Should skip if repo exists, but is not connected", () => {
+        const handler = new tabEventHandler(repoPath);
+        handler.handleTabChangeEvent();
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)    
+    })
+
+    // Skip case where repo is connected, but user account is not valid
+    test("Should skip if repo is connected, but user account is not valid", () => {
+        configUtil.addRepo();
+        addUser(baseRepoPath, false);
+        const handler = new tabEventHandler(repoPath);
+        handler.handleTabChangeEvent();
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)       
+    })
+
+    // Skip case where user account is not valid, but repo is not connected 
+    test("Should skip if user account is valid, but repo is not connected", () => {
+        addUser(baseRepoPath, true);
+        const handler = new tabEventHandler(repoPath);
+        handler.handleTabChangeEvent();
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)      
+    })
+
+    // Skip case where file is changed
+    test("Should skip if file is changed", () => {
+        // Repo is connected & user account is also valid
+        configUtil.addRepo();
+        addUser(baseRepoPath, true);
+        const handler = new tabEventHandler(repoPath);
+        // Pass isTabEvent=false, which means changeEvent.changed.length > 0
+        handler.handleTabChangeEvent(false);
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)    
+    })
+
+    // Skip if invalid repo id
+    test("Should skip is repoId doesn't exist", () => {
+        const invalid_repo_path = "/home/documents"
+        const configUtil_2 = new Config(invalid_repo_path, configPath);
+        configUtil.addRepo();
```

Comment:
Why do we need to add a repo in this test when we are testing the case of repoPath not existing in the config

Reply:
Was testing with a different repo id, that wasn't synced - since that would not have a repo id associated. 

We can remove `configUtil.addRepo();` , but then our code will return on line `if(!this.repoPath) return;`

Reply:
repoPath is the path of currently opened repo, does not need to be in config file

Reply:
Right?